Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -48,41 +63,89 @@ void APSStarActor::OnLocalCharacterReady(APlayerCharacter* Character, int32 Char
 	}
 }
 
+// Called when the current game state was changed
+void APSStarActor::OnGameStateChanged(ECurrentGameState GameState)
+{
+	if (AMyGameStateBase::Get().GetCurrentGameState() == ECurrentGameState::Menu)
+	{
+		SetStartTimeMenuStars();
+		TryPlayMenuStarAnimation();
+	}
+	else
+	{
+		SetActorTickEnabled(false);
+		StartTimeMenuStarsInternal = 0.f;
+	}
+}
+
 // Is called when any cinematic started
 void APSStarActor::OnAnyCinematicStarted(const UObject* LevelSequence, const UObject* FromInstigator)
 {
-	const UWorld* World = GetWorld();
-	check(World);
-
-	StartTimeHideStarsInternal = World->GetTimeSeconds();
-
-	// Enable tick and process this frame
-	SetActorTickEnabled(true);
+	SetStartTimeHideStars();
 	TryPlayHideStarAnimation();
 }
 
-// Hiding stars with animation
+// Hiding stars with animation in main menu when cinematic is start to play
 void APSStarActor::TryPlayHideStarAnimation()
 {
-	if (!StartTimeHideStarsInternal)
+	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
+	const bool bIsFinished = !TryPlayStarAnimation(StartTimeHideStarsInternal, CurrentRow.HideStarsAnimation);
+	if (bIsFinished)
 	{
-		return;
+		StartTimeHideStarsInternal = 0.f;
+		SetActorTickEnabled(false);
```

Comment:
1. Please return `this` star to the Pool at this line, only when Finished hiding the star
2. Remove all `SetActorTickEnabled` everywhere in this file as now 100% of lifetime the stars should be ticking (In Menu + hide). All the rest time stars will be returned to the Pool, meaning Pool will disable tick for you automatically
```
UPoolManagerSubsystem::Get().ReturnToPool(this);
```

Reply:
done as part of https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/8639210197ed6948fb7113ecbb6ed9372c267855

Reply:
@h4rdmol I still can find `SetActorTickEnabled(true)` in both Setters, you don't need enable tick anymore as it's always ticking now while active since it's always playing some animation.

Please remove the rest of `SetActorTickEnabled(true)` in your Setters
![image](https://github.com/user-attachments/assets/76a50e27-5ed0-4a44-b248-c813794d7b9c)

Reply:
my bad, achieved her https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/df58201938adfe61a41317402f8c79a7dd406cce